src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        linkHead(value);
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> node = new Node<>(null, value, null);
+
+        Node<T> current = getNodeAtIndexAddition(index);
+        if (index == 0) {
+            node.next = head;
+            if (head != null) {
+                head.prev = node;
+            }
+            head = node;
+            if (tail == null) {
+                tail = node;
+            }
+        } else if (index == size()) {
+            if (tail == null) {
+                head = node;
+            } else {
+                linkTail(node);
+            }
+        } else {
+            node.next = current;
+            node.prev = current.prev;
+            if (current.prev != null) {
+                current.prev.next = node;
+            }
+            current.prev = node;
+            node.size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> current = getNodeAtIndex(index);
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> current = getNodeAtIndex(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getNodeAtIndex(index);
+        baseRemove(current);
+
+        return current.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if (object == null) {
+                if (current.value == null) {
+                    baseRemove(current);
+                    return true;
+                }
+            } else if (object.equals(current.value)) {
+                baseRemove(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        Node<T> current = head;
+        int size = 0;
+        while (current != null) {
+            size++;
+            current = current.next;
+        }
+        return size;
```

Comment:
this is bad solution, cause not optimizing performance. 

make field `size` and no need to calculate it each time in method size

```suggestion
        Node<T> current = head;
        int size = 0;
        while (current != null) {
            size++;
            current = current.next;
        }
        return size;
```